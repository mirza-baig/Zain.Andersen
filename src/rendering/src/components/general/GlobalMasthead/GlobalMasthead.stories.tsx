// Global
import { Story, Meta } from '@storybook/react';
// Local
import GlobalMasthead, { GlobalMastheadProps } from './GlobalMasthead';
import defaultData from './GlobalMasthead.mock-data';

export default {
  title: 'General/GlobalMasthead',
  component: GlobalMasthead,
} as Meta;
const Template: Story<GlobalMastheadProps> = (props) => <GlobalMasthead {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
