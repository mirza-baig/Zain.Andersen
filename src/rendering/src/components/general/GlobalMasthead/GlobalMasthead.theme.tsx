// Lib
import { ThemeFile } from 'lib/context/ThemeContext';

export const GlobalMastheadTheme: ThemeFile = {
  aw: {
    classes: {
      headline: 'text-sm-m ml:text-m  font-heavy',
      headWrapper: 'w-full flex flex-col ml:items-center ml:justify-center py-4 px-5',
      headLogoWrapper: 'flex items-center justify-start ml:justify-center',
      headLogo: 'flex items-center',
      mastheadWrapper:
        'theme-black sticky top-[55px] left-0 right-0 z-[15] col-span-12 flex place-items-center justify-center ml:top-[122px]',
      menuIcon: 'ml:hidden mr-5',
      anchorWrapper:
        'mt-4 flex flex-col ml:flex-row items-start ml:items-center justify-between w-full transition ease-in-out delay-150',
      socialIconsWrapper: 'mt-4 ml:mt-0',
      iconWrapper: 'flex items-center space-x-2',
      socialIcon: '',
      anchors:
        'lg:space-x-[80px] ml:space-x-[50px] space-y-[12px] ml:space-y-0 flex flex-col ml:flex-row font-semibold',
      linkTitle: 'text-[18px] font-sans font-bold text-white',
    },
  },
  rba: {
    classes: {
      headline: 'text-white text-sm-m ml:text-m  font-heavy',
      headWrapper: 'w-full flex flex-col ml:items-center ml:justify-center py-4 px-5',
      headLogoWrapper: 'flex items-center justify-start ml:justify-center',
      headLogo: 'flex items-center',
      mastheadWrapper:
        'theme-black sticky top-[55px] left-0 right-0 z-[99] col-span-12 flex place-items-center justify-center ml:top-[122px]',
      menuIcon: 'ml:hidden mr-5',
      anchorWrapper:
        'mt-4 flex flex-col ml:flex-row items-start ml:items-center justify-between w-full transition ease-in-out delay-150',
      socialIconsWrapper: 'mt-4 ml:mt-0',
      iconWrapper: 'flex items-center space-x-4',
      socialIcon: '',
      anchors: 'ml:space-x-[16px] space-y-[12px] ml:space-y-0 flex flex-col ml:flex-row',
      linkTitle: 'text-[16px] text-bold font-sans font-semibold text-white',
    },
  },
};
