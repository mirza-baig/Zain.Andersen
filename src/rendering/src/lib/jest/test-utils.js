// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
