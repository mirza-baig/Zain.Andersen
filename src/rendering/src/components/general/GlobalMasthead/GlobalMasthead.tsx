// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { useTheme } from 'lib/context/ThemeContext';
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
// Components
import classNames from 'classnames';
import { GlobalMastheadTheme } from './GlobalMasthead.theme';
import { useMemo, useState, useEffect, useRef } from 'react';
import ImageWrapper from 'src/helpers/Media/ImageWrapper';
import { FiArrowRight } from 'react-icons/fi';
import { IoIosArrowDropdown, IoIosArrowDropup } from 'react-icons/io';
import { useCurrentScreenType, getBreakpoint } from 'lib/utils/get-screen-type';
import { getEnum } from 'lib/utils';
import SocialIcons from './SocialIcons.helper';
import { LinkWrapper } from 'src/helpers/LinkWrapper';
import { Foundation } from 'src/.generated/Foundation.EnterpriseWeb.model';
import { Item } from '@sitecore-jss/sitecore-jss-nextjs';

export type BackgroundColor = 'gray' | 'primary' | 'white';

export type GlobalMastheadProps =
  Feature.EnterpriseWeb.AndersenWindows.Components.General.GlobalMasthead.GlobalMasthead;
const GlobalMasthead = (props: GlobalMastheadProps) => {
  const selectedBackgroundColor =
    getEnum<BackgroundColor>(props.fields?.backgroundColor) || 'white';

  const { themeData } = useTheme(GlobalMastheadTheme);
  const { currentScreenWidth } = useCurrentScreenType();
  const mastheadRef = useRef<HTMLDivElement>(null);

  const isDesktop = currentScreenWidth >= getBreakpoint('ml') ? true : false;

  const DesktopLogoImage = props.fields?.desktopLogo;
  const MobileLogoImage = props.fields?.mobileLogo?.value?.src
    ? props.fields.mobileLogo
    : props.fields?.desktopLogo;

  const backgroundColorStyle = useMemo(() => {
    if (selectedBackgroundColor === 'gray') {
      return 'bg-gray';
    } else if (selectedBackgroundColor === 'primary') {
      return 'bg-primary';
    } else {
      return 'bg-white';
    }
  }, [selectedBackgroundColor]);

  const fontTextStyle = useMemo(() => {
    if (selectedBackgroundColor === 'white') {
      return 'text-black';
    } else {
      return 'text-white';
    }
  }, [selectedBackgroundColor]);

  const iconColorStyle = useMemo(() => {
    if (selectedBackgroundColor?.toLowerCase() === 'white') {
      return 'black';
    } else {
      return 'white';
    }
  }, [selectedBackgroundColor]);

  const [isShow, setIsShow] = useState(false);

  useEffect(() => {
    setIsShow(isDesktop);
  }, [isDesktop]);

  return (
    <div
      data-component="general/globalmasthead"
      className={classNames(themeData.classes.mastheadWrapper, backgroundColorStyle)}
      ref={mastheadRef}
    >
      <div className={themeData.classes.headWrapper}>
        <div className={classNames(themeData.classes.headLogoWrapper, fontTextStyle)}>
          <div onClick={() => setIsShow(!isShow)} className={themeData.classes.menuIcon}>
            {!isShow ? <IoIosArrowDropdown size={36} /> : <IoIosArrowDropup size={36} />}
          </div>
          <div className={themeData.classes.headLogo}>
            <LinkWrapper
              field={props.fields?.linkLogo}
              suppressLinkText
              ariaLabel={{
                value: isDesktop
                  ? DesktopLogoImage?.value?.text
                  : MobileLogoImage?.value?.text || 'global masthead Logo Image',
              }}
            >
              {isDesktop ? (
                <>
                  {DesktopLogoImage?.value?.src ? (
                    <div className={`h-[60px] w-[550px] object-cover`}>
                      <ImageWrapper
                        image={DesktopLogoImage}
                        mobileImage={MobileLogoImage}
                        additionalDesktopClasses="h-full w-full cursor-pointer"
                      />
                    </div>
                  ) : (
                    <span className={themeData.classes.headline}>
                      {props.fields?.headlineText.value}
                    </span>
                  )}
                </>
              ) : (
                <>
                  {MobileLogoImage?.value?.src ? (
                    <div className={`h-[30px] w-[300px] object-cover`}>
                      <ImageWrapper
                        image={MobileLogoImage}
                        mobileImage={MobileLogoImage}
                        additionalDesktopClasses="h-full w-full cursor-pointer"
                      />
                    </div>
                  ) : (
                    <span className={themeData.classes.headline1}>
                      {props.fields?.headlineText.value}
                    </span>
                  )}
                </>
              )}
            </LinkWrapper>
          </div>
        </div>
        {isShow && (
          <div className={themeData.classes.anchorWrapper}>
            <div className="hidden ml:block">
              <SocialIcons icons={props?.fields?.socialIcons} iconColor={iconColorStyle} />
            </div>
            <div className={themeData.classes.anchors}>
              {props.fields?.children.map(
                (link: Item & Foundation.EnterpriseWeb.Enterprise.FieldSets.GeneralLink) => (
                  <LinkWrapper
                    field={link?.fields?.Link}
                    key={link?.id}
                    suppressLinkText
                    ariaLabel={{ value: (link.fields.Link.value.text as string) || 'General Link' }}
                  >
                    <span className={classNames(themeData.classes.linkTitle, fontTextStyle)}>
                      {link.fields.Link.value.text}
                    </span>
                  </LinkWrapper>
                )
              )}
            </div>
            <div className="mt-2 py-5 pb-2 mmd:py-2 ml:mt-0">
              <LinkWrapper
                field={props.fields?.rightSideLink}
                suppressLinkText
                className={classNames('flex items-center', fontTextStyle)}
                ariaLabel={{ value: props.fields?.rightSideLink.value.text || 'Right Side Link' }}
              >
                <span className="font-sans text-[18px] font-bold">
                  {props.fields?.rightSideLink.value.text}
                </span>
                <FiArrowRight size={16} />
              </LinkWrapper>
            </div>
            <div className="block ml:hidden">
              <SocialIcons icons={props?.fields?.socialIcons} iconColor={iconColorStyle} />
            </div>
          </div>
        )}
      </div>
    </div>
  );
};

export default withDatasourceCheck()<GlobalMastheadProps>(GlobalMasthead);
