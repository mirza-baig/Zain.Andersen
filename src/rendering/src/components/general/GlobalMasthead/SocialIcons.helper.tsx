import { useTheme } from 'lib/context/ThemeContext';
import { GlobalMastheadTheme } from './GlobalMasthead.theme';
import {
  FaFacebook,
  FaTwitter,
  FaPinterest,
  FaInstagram,
  FaLinkedin,
  FaYoutube,
  FaHouzz,
} from 'react-icons/fa';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { getEnum } from 'lib/utils';

const SocialIcons = ({
  icons,
  iconColor,
}: {
  icons: Feature.EnterpriseWeb.Enterprise.Elements.Social.SocialLink[];
  iconColor: string;
}) => {
  const { themeData } = useTheme(GlobalMastheadTheme);

  return (
    <div className={themeData.classes.socialIconsWrapper}>
      <div className={themeData.classes.iconWrapper}>
        {icons &&
          icons.map((icon) => (
            <a
              href={icon?.fields?.link?.value?.href}
              target="_blank"
              key={icon.id}
              rel="noreferrer"
            >
              {getEnum(icon?.fields?.icon) === 'facebook' && (
                <FaFacebook size={20} color={iconColor} />
              )}
              {getEnum(icon?.fields?.icon) === 'twitter' && (
                <FaTwitter size={20} color={iconColor} />
              )}
              {getEnum(icon?.fields?.icon) === 'pinterest' && (
                <FaPinterest size={20} color={iconColor} />
              )}
              {getEnum(icon?.fields?.icon) === 'instagram' && (
                <FaInstagram size={20} color={iconColor} />
              )}
              {getEnum(icon?.fields?.icon) === 'linkedin' && (
                <FaLinkedin size={20} color={iconColor} />
              )}
              {getEnum(icon?.fields?.icon) === 'youtube' && (
                <FaYoutube size={20} color={iconColor} />
              )}
              {getEnum(icon?.fields?.icon) === 'houzz' && <FaHouzz size={20} color={iconColor} />}
            </a>
          ))}
      </div>
    </div>
  );
};

export default SocialIcons;
