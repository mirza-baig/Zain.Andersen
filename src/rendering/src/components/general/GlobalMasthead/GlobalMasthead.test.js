// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import GlobalMasthead from './GlobalMasthead';
import defaultData from './GlobalMasthead.mock-data';

it('renders correctly', async () => {
  const component = snapshot(GlobalMasthead, { componentProps: defaultData });
  await hasDataComponent(component, 'general/globalmasthead');
});
