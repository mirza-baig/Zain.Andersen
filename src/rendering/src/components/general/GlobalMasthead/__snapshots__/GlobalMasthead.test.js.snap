// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="theme-black sticky top-[55px] left-0 right-0 z-[15] col-span-12 flex place-items-center justify-center ml:top-[122px] bg-gray"
  data-component="general/globalmasthead"
>
  <div
    class="w-full flex flex-col ml:items-center ml:justify-center py-4 px-5"
  >
    <div
      class="flex items-center justify-start ml:justify-center text-white"
    >
      <div
        class="ml:hidden mr-5"
      >
        <svg
          fill="currentColor"
          height="36"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="36"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M147.6 301.3c-7.5-7.5-7.5-19.8 0-27.3l95.7-95.4c7.3-7.3 19.1-7.5 26.6-.6l94.3 94c3.8 3.8 5.7 8.7 5.7 13.7 0 4.9-1.9 9.9-5.6 13.6-7.5 7.5-19.7 7.6-27.3 0l-81-79.8-81.1 81.9c-7.5 7.5-19.7 7.5-27.3-.1z"
          />
          <path
            d="M256 464c114.9 0 208-93.1 208-208S370.9 48 256 48 48 141.1 48 256s93.1 208 208 208zm0-32c-47 0-91.2-18.3-124.4-51.6C98.3 347.2 80 303 80 256s18.3-91.2 51.6-124.4C164.8 98.3 209 80 256 80s91.2 18.3 124.4 51.6C413.7 164.8 432 209 432 256s-18.3 91.2-51.6 124.4C347.2 413.7 303 432 256 432z"
          />
        </svg>
      </div>
      <div
        class="flex items-center"
      >
        <a
          aria-label=""
          class=""
          data-gtm-click=""
          data-gtm-dl-event="cta_click"
          href="/adam-test/"
          role="link"
          tabindex="0"
          target=""
          title=""
        >
          
          <div
            class="h-[60px] w-[550px] object-cover"
          >
            <div
              class="relative hidden md:block aspect-picture    h-full w-full cursor-pointer"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 8.0625% 0px 0px 0px;"
                />
                <img
                  alt="pocketing"
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="relative block w-full md:hidden"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 8.0625% 0px 0px 0px;"
                />
                <img
                  alt="pocketing"
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
                />
                <noscript />
              </span>
            </div>
          </div>
        </a>
      </div>
    </div>
    <div
      class="mt-4 flex flex-col ml:flex-row items-start ml:items-center justify-between w-full transition ease-in-out delay-150"
    >
      <div
        class="hidden ml:block"
      >
        <div
          class="mt-4 ml:mt-0"
        >
          <div
            class="flex items-center space-x-2"
          >
            <a
              href="https://www.facebook.com/AndersenWindows?utm_source=andersenwindows.com&utm_medium=referral"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                color="white"
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 512 512"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
                />
              </svg>
            </a>
            <a
              href="https://twitter.com/andersenwindows"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                color="white"
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 512 512"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                />
              </svg>
            </a>
            <a
              href="https://twitter.com/andersenwindows"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                color="white"
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 496 512"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M496 256c0 137-111 248-248 248-25.6 0-50.2-3.9-73.4-11.1 10.1-16.5 25.2-43.5 30.8-65 3-11.6 15.4-59 15.4-59 8.1 15.4 31.7 28.5 56.8 28.5 74.8 0 128.7-68.8 128.7-154.3 0-81.9-66.9-143.2-152.9-143.2-107 0-163.9 71.8-163.9 150.1 0 36.4 19.4 81.7 50.3 96.1 4.7 2.2 7.2 1.2 8.3-3.3.8-3.4 5-20.3 6.9-28.1.6-2.5.3-4.7-1.7-7.1-10.1-12.5-18.3-35.3-18.3-56.6 0-54.7 41.4-107.6 112-107.6 60.9 0 103.6 41.5 103.6 100.9 0 67.1-33.9 113.6-78 113.6-24.3 0-42.6-20.1-36.7-44.8 7-29.5 20.5-61.3 20.5-82.6 0-19-10.2-34.9-31.4-34.9-24.9 0-44.9 25.7-44.9 60.2 0 22 7.4 36.8 7.4 36.8s-24.5 103.8-29 123.2c-5 21.4-3 51.6-.9 71.2C65.4 450.9 0 361.1 0 256 0 119 111 8 248 8s248 111 248 248z"
                />
              </svg>
            </a>
            <a
              href="https://www.Huzz.com/"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                color="white"
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 448 512"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M275.9 330.7H171.3V480H17V32h109.5v104.5l305.1 85.6V480H275.9z"
                />
              </svg>
            </a>
            <a
              href="https://www.youtube.com/"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                color="white"
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 576 512"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
      <div
        class="lg:space-x-[80px] ml:space-x-[50px] space-y-[12px] ml:space-y-0 flex flex-col ml:flex-row font-semibold"
      >
        <a
          aria-label="All Post-1"
          class=""
          data-gtm-click=""
          data-gtm-dl-event="cta_click"
          href="/components/"
          role="link"
          tabindex="0"
          title=""
        >
          
          <span
            class="text-[18px] font-sans font-bold text-white text-white"
          >
            All Post-1
          </span>
        </a>
        <a
          aria-label="Company News"
          class=""
          data-gtm-click=""
          data-gtm-dl-event="cta_click"
          href="/components/"
          role="link"
          tabindex="0"
          title=""
        >
          
          <span
            class="text-[18px] font-sans font-bold text-white text-white"
          >
            Company News
          </span>
        </a>
        <a
          aria-label="Comunicate"
          class=""
          data-gtm-click=""
          data-gtm-dl-event="cta_click"
          href="/components/"
          role="link"
          tabindex="0"
          title=""
        >
          
          <span
            class="text-[18px] font-sans font-bold text-white text-white"
          >
            Comunicate
          </span>
        </a>
        <a
          aria-label="Innovation"
          class=""
          data-gtm-click=""
          data-gtm-dl-event="cta_click"
          href="/components/"
          role="link"
          tabindex="0"
          title=""
        >
          
          <span
            class="text-[18px] font-sans font-bold text-white text-white"
          >
            Innovation
          </span>
        </a>
        <a
          aria-label="General Link"
          data-gtm-click=""
          data-gtm-dl-event="cta_click"
          href=""
          role="link"
          tabindex="0"
        >
          
          <span
            class="text-[18px] font-sans font-bold text-white text-white"
          />
        </a>
      </div>
      <div
        class="mt-2 py-5 pb-2 mmd:py-2 ml:mt-0"
      >
        <a
          aria-label="As seen"
          class="flex items-center text-white"
          data-gtm-click=""
          data-gtm-dl-event="cta_click"
          href="/demo-herotwocolumn/"
          role="link"
          tabindex="0"
          title=""
        >
          
          <span
            class="font-sans text-[18px] font-bold"
          >
            As seen
          </span>
          <svg
            fill="none"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="5"
              x2="19"
              y1="12"
              y2="12"
            />
            <polyline
              points="12 5 19 12 12 19"
            />
          </svg>
        </a>
      </div>
      <div
        class="block ml:hidden"
      >
        <div
          class="mt-4 ml:mt-0"
        >
          <div
            class="flex items-center space-x-2"
          >
            <a
              href="https://www.facebook.com/AndersenWindows?utm_source=andersenwindows.com&utm_medium=referral"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                color="white"
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 512 512"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
                />
              </svg>
            </a>
            <a
              href="https://twitter.com/andersenwindows"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                color="white"
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 512 512"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                />
              </svg>
            </a>
            <a
              href="https://twitter.com/andersenwindows"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                color="white"
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 496 512"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M496 256c0 137-111 248-248 248-25.6 0-50.2-3.9-73.4-11.1 10.1-16.5 25.2-43.5 30.8-65 3-11.6 15.4-59 15.4-59 8.1 15.4 31.7 28.5 56.8 28.5 74.8 0 128.7-68.8 128.7-154.3 0-81.9-66.9-143.2-152.9-143.2-107 0-163.9 71.8-163.9 150.1 0 36.4 19.4 81.7 50.3 96.1 4.7 2.2 7.2 1.2 8.3-3.3.8-3.4 5-20.3 6.9-28.1.6-2.5.3-4.7-1.7-7.1-10.1-12.5-18.3-35.3-18.3-56.6 0-54.7 41.4-107.6 112-107.6 60.9 0 103.6 41.5 103.6 100.9 0 67.1-33.9 113.6-78 113.6-24.3 0-42.6-20.1-36.7-44.8 7-29.5 20.5-61.3 20.5-82.6 0-19-10.2-34.9-31.4-34.9-24.9 0-44.9 25.7-44.9 60.2 0 22 7.4 36.8 7.4 36.8s-24.5 103.8-29 123.2c-5 21.4-3 51.6-.9 71.2C65.4 450.9 0 361.1 0 256 0 119 111 8 248 8s248 111 248 248z"
                />
              </svg>
            </a>
            <a
              href="https://www.Huzz.com/"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                color="white"
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 448 512"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M275.9 330.7H171.3V480H17V32h109.5v104.5l305.1 85.6V480H275.9z"
                />
              </svg>
            </a>
            <a
              href="https://www.youtube.com/"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                color="white"
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 576 512"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
